django/db/migrations/state.py

```diff
@@ -205,8 +234,29 @@ def alter_field(self, app_label, model_name, name, field, preserve_default):
             field.default = NOT_PROVIDED
         else:
             field = field
-        model_state = self.models[app_label, model_name]
-        model_state.fields[name] = field
+        model_key = (app_label, model_name)
+        model_state = self.models[model_key]
+        fields = model_state.fields
+        if self._relations is not None:
+            old_remote_field = fields[name].remote_field
+            new_remote_field = field.remote_field
+            if old_remote_field and new_remote_field:
```

Comment:
I think this would be simpler if you did:

```python
if old_remote_field:
    if new_remote_field:
        ...
    else:
        # Include code from the `if old_remote_field` case below.
        ...
elif new_remote_field:
        # Include code from the `elif new_remote_field` case below.
```

Reply:
Committed the change 👍